addons/alerting.libsonnet

```diff
@@ -1,205 +1,117 @@
-    "If 'alerting' is set, 'slackWebhookURLCritical' or 'pagerdutyRoutingKey' should be declared"
-  ),
-
-  local pdConfig =
-    |||
-      pagerduty_configs:
-        - send_resolved: true
-          routing_key: '%(pagerdutyRoutingKey)s'
-    |||,
-
-  local slackConfig =
-    |||
-      slack_configs:
-        - send_resolved: true
-          api_url: %(webhookURL)s
-          channel: '%(slackChannel)s'
-          title: '[{{ .Status | toUpper }}{{ if eq .Status "firing" }}{{ end }}] %(clusterName)s Monitoring'
-          text: |
-            {{ range .Alerts }}
-            **Please take immediate action!**
-            *Cluster:* {{ .Labels.cluster }}
-            *Alert:* {{ .Labels.alertname }}
-            *Description:* {{ .Annotations.description }}
-            {{ end }}
-          actions:
-          - type: button
-            text: 'Runbook :book:'
-            url: '{{ .CommonAnnotations.runbook_url }}'
-    |||,
-
-  local globalPagerDutyRoutingKey = if std.objectHas(config.alerting, 'pagerdutyRoutingKey')
-  then config.alerting.pagerdutyRoutingKey
-  else '',
-
-  local globalSlackWebhookURL = if std.objectHas(config.alerting, 'slackWebhookURLCritical')
-  then config.alerting.slackWebhookURLCritical
-  else '',
-
-  local globalCriticalReceiver =
-    if std.length(globalPagerDutyRoutingKey) > 0
-    then
-      pdConfig % {
-        pagerdutyRoutingKey: globalPagerDutyRoutingKey,
-      }
-    else
-      slackConfig % {
-        clusterName: config.clusterName,
-        webhookURL: globalSlackWebhookURL,
-        slackChannel: config.alerting.slackChannelPrefix + '_critical',
+  local teamWebHookMap = [
+    {
+      team: 'platform',
+      webhook: config.alerting.platform.slackWebhookURL,
+      //pd_key: config.alerting.platform.pagerdutyRoutingKey,
+    },
+    {
+      team: 'ide',
+      webhook: config.alerting.IDE.slackWebhookURL,
+      //pd_key: config.alerting.IDE.slackWebhookURL,
+    },
+    {
+      team: 'workspace',
+      webhook: config.alerting.workspace.slackWebhookURL,
+      //pd_key: config.alerting.workspaces.slackWebhookURL,
+    },
+    {
+      team: 'webapp',
+      webhook: config.alerting.webapp.slackWebhookURL,
+      //pd_key: config.alerting.webapp.slackWebhookURL,
+    },
+  ],
+
+  local routeTmpl = |||
+    - receiver: %(team)sTeamReceiver
+      match:
+        team: %(team)s
+  |||,
+
+  local routeCriticalTmpl = |||
+    - receiver: %(team)sTeamCriticalReceiver
+      match:
+        team: %(team)s
+        severity: %(severity)s
+  |||,
+
+  local teamRoutesTmpls = [
+    if std.objectHas(p, 'team') then routeTmpl % {
+      team: p.team,
+    }
+    for p in teamWebHookMap
+  ],
+
+  local teamCriticalRoutesTmpls = [
+    if std.objectHas(p, 'team') && std.objectHas(p, 'pd_key') then routeCriticalTmpl % {
+      team: p.team,
+      severity: 'critical',
+    }
+    for p in teamWebHookMap
+  ],
```

Comment:
This part of the code won't be used at the moment, because we agreed that all critical alerts go to the company-wide pagerduty schedule.

Keeping code that isn't used is dangerous 😬, it can break without us noticing exactly because we don't use it. Let's re-introduce this once the company agrees with per-team on-call schedules